### PR TITLE
Update hypre to 2.22.1 and CMake to 3.16

### LINKED ISSUE #431: Clean for next hypre update
see https://github.com/ECP-copa/Cabana/pull/426#discussion_r662511399

Next hypre update we can drop `enable_language(C)`.

### PR DESCRIPTION
**Pull request by streeve**
Remove workaround with new tagged release

Fixes #431

### CODE REVIEW

**Review by junghans — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,7 +83,8 @@ Cabana_add_dependency( PACKAGE ArborX )
 find_package( CLANG_FORMAT 10 )
 
 # find hypre
-Cabana_add_dependency( PACKAGE HYPRE VERSION 2.22.0 )
+Cabana_add_dependency( PACKAGE HYPRE VERSION 2.22.1 )
+message("Debug ${HYPRE_DIR}")
```
Comment (dalg24) [inline CMakeLists.txt:87]: Remove

File: docker/Dockerfile [hunk 2]
```diff
@@ -161,4 +161,5 @@ RUN HYPRE_VERSION=v2.22.0 && \
       -D HYPRE_WITH_MPI=ON \
     ../src && \
     make -j${NPROCS} install && \
+    cat /opt/hypre/lib/cmake/HYPRE/HYPREConfig.cmake && \
```
Comment (dalg24) [inline docker/Dockerfile:164]: Remove

**Review by dalg24 — CHANGES_REQUESTED** [review]
Drop debug print statements

**Comment by dalg24** [comment]
For reference
https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Cabana/detail/PR-446/7/pipeline#step-48-log-976

**Comment by junghans** [comment]
@dalg24 calm yourself, we are still debugging this.....

**Comment by dalg24** [comment]
> @dalg24 calm yourself, we are still debugging this.....

Got it :)

This is weird indeed.  The link I posted is to you debug prints.  `HYPRE_WITH_MPI` is `TRUE`.  Do you know how to make the `find_dependency(MPI ...)` verbose to make sure it does not bypass it with our having already found MPI?

**Comment by dalg24** [comment]
(I blocked to make sure auto-merge does not happen)

**Review by junghans — APPROVED** [review]

**Comment by junghans** [comment]
@sslattery @dalg24 @streeve turns out this was a bug in CMake, bumping it to 3.16 helped. 
Current Kokkos master requires 3.16 as well, so not a great restriction, as we will require kokkos-3.3 at some point anyhow.

**Comment by dalg24** [comment]
> @sslattery @dalg24 @streeve turns out this was a bug in CMake, bumping it to 3.16 helped.
> Current Kokkos master requires 3.16 as well, so not a great restriction, as we will require kokkos-3.3 at some point anyhow.

Is it FindMPI module bug in older versions as I suspected earlier on the phone?

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
We need to bump our cmake requirement on a follow up pr

File: CMakeLists.txt [hunk 3]
```diff
@@ -12,17 +12,11 @@
 #------------------------------------------------------------------------------#
 # Project settings
 #------------------------------------------------------------------------------#
-cmake_minimum_required(VERSION 3.9)
+cmake_minimum_required(VERSION 3.16)
```
Comment (junghans) [inline CMakeLists.txt:15]: @dalg24 I bumped it here.

**Pull request merged**